analyzers/src/SonarAnalyzer.CSharp/Rules/MultilineBlocksWithoutBrace.cs

```diff
@@ -28,143 +28,118 @@
+            var secondPosition = second.GetLocation().GetLineSpan().StartLinePosition;
+            var rootPosition = context.Node.GetLocation().GetLineSpan().StartLinePosition;
 
-            var nextStatement = context.Node.GetLastToken().GetNextToken().Parent;
-            // This algorithm to get the next statement can sometimes return a parent statement (for example a BlockSyntax)
-            // so we need to filter this case by returning if the nextStatement happens to be one ancestor of statement.
-            if (nextStatement == null ||
-                statement.Ancestors().Contains(nextStatement))
+            if (SameIndentation(firstPosition, secondPosition)
+                || DifferentIndentation(rootPosition, secondPosition))
             {
-                return;
-            }
-
-            var statementPosition = statement.GetLocation().GetLineSpan().StartLinePosition;
-            var nextStatementPosition = nextStatement.GetLocation().GetLineSpan().StartLinePosition;
-
-            if (statementPosition.Character == nextStatementPosition.Character)
-            {
-                var lineSpan = context.Node.SyntaxTree.GetText().Lines[nextStatementPosition.Line].Span;
-                var location = Location.Create(context.Node.SyntaxTree, TextSpan.FromBounds(nextStatement.SpanStart, lineSpan.End));
-
-                context.ReportIssue(rule.CreateDiagnostic(context.Compilation,
-                    location,
-                    additionalLocations: new[] { statement.GetLocation() },
-                    messageArgs: new object[] { executed, execute, nextStatementPosition.Line - statementPosition.Line + 1 }));
+                var lineSpan = context.Node.SyntaxTree.GetText().Lines[secondPosition.Line].Span;
+                var location = Location.Create(context.Node.SyntaxTree, TextSpan.FromBounds(second.SpanStart, lineSpan.End));
+                var blockSize = secondPosition.Line - firstPosition.Line + 1;
+                var additional = new[] { first.GetLocation() };
+                context.ReportIssue(Rule.CreateDiagnostic(context.Compilation, location, additional, executed, execute, blockSize));
             }
         }
-
-        private static bool IsStatementCandidateLoop(StatementSyntax statement) =>
-            statement.IsKind(SyntaxKind.ForEachStatement) ||
-            statement.IsKind(SyntaxKind.ForStatement) ||
-            statement.IsKind(SyntaxKind.WhileStatement);
     }
+
+    private static bool SameIndentation(LinePosition first, LinePosition second) =>
+        first.Character == second.Character;
+
+    private static bool DifferentIndentation(LinePosition first, LinePosition second) =>
+        first.Character != second.Character;
+
+    private static SyntaxNode SecondStatement(SyntaxNode root, SyntaxNode first) =>
+        !first.IsKind(SyntaxKind.Block)
+        && root.GetLastToken().GetNextToken().Parent is { } second
+        && !first.Ancestors().Contains(second)
```

Comment:
There used to be a comment like this that explains the weirdness of the check. Let's preserve it - or make it shorter to fit on the `Contains` line
```suggestion
        // This algorithm to get the next statement can sometimes return a parent statement (for example a BlockSyntax)
        // so we need to filter this case by returning if the nextStatement happens to be one ancestor of statement.
        && root.GetLastToken().GetNextToken().Parent is { } second
        && !first.Ancestors().Contains(second)
```